src/useQueryParam.ts

```diff
@@ -31,11 +33,18 @@ export const useQueryParam = <D, D2 = D>(
 ): [D2 | undefined, (newValue: D, updateType?: UrlUpdateType) => void] => {
   const { history, location } = React.useContext(QueryParamContext);
 
-  // read in the raw query
   if (!rawQuery) {
-    rawQuery = React.useMemo(() => parseQueryString(location.search) || {}, [
-      location.search,
-    ]);
+    rawQuery = React.useMemo(() => {
+      let pathname = {}
+
+      if (typeof location === 'object' && typeof window !== undefined) {
```

Comment:
Shouldn't it be typeof `window !== "undefined"` ?

Reply:
Also, can you explain why this fixes the problem? I don't see `window` being referenced in this code.

Reply:
You're right, we should check `typeof window !== 'undefined', fixed that.
We're only checking if the code is running in the browser or not.